docs/extensions.rst

```diff
@@ -77,3 +77,37 @@ and treat it as if the client had made a request.
 
 The ASGI server should set the pseudo ``:authority`` header value to
 be the same value as the request that triggered the push promise.
+
+Sendfile
+------------------
+
+Sendfile allows you to send the contents of a file descriptor to the
+HTTP client with zero copy. ASGI servers that implement this extension will
+provide ``http.response.sendfile`` in the extensions part of the scope::
+
+    "scope": {
+        ...
+        "extensions": {
+            "http.response.sendfile": {},
+        },
+    }
+
+An ASGI framework can call ``sendfile`` by sending a message with
+the following keys. This message can be sent at any time after the
+*Response Start* message but before the final *Response Body* message.
```

Comment:
Should we say that it's not allowed to send a response body if you also use this?

Reply:
Zero-copy transmission data should not be special, it should be mixed with `http.response.body`.

Reply:
Ah, interesting. That should be made very clear here, then - that servers should send this data exactly between any other `response.body` messages.